### ISSUE TITLE
CA1034: Allow Nested Static Types

### ISSUE THREAD

**Issue by manfred-brands** [dotnet/roslyn-analyzers#3693]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)

#### Diagnostic ID

Example: [CA1034](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1034?view=vs-2019)

#### Repro steps

```csharp
    public static class WellKnown
    {
        public enum Values
        {
            Input = 0,
            Output = 1,
        }

        public static class Names
        {
            public const string Input = "Input";
            public const string Output = "Output";
        }
    }
```

#### Wanted behavior

CA1034 does not fire on either nested type.

#### Actual behavior

CA1034 does allow 'WellKnown.Values.Input', but not 'WellKnown.Names.Input'.
Both are equivalent. 
A static type complies with "The Framework Design CodingGuidelines (section 4.9)" as it can neither be referenced nor instantiated.

If you agree in allowing the latter, I can raise a pull request as I already made the changes.

**Comment by Evangelink** [comment]
Looking at the code it seems that the enum exception is for compatibility with the [old fxcop behavior](https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/NestedTypesShouldNotBeVisible.cs#L99-L101). So if we want to start reporting on them we will need to introduce a user-option to ensure we still provide the old behavior. Besides, the rule seems to be marked as candidate for removal so I am not sure if @mavasani is expecting much work on it.

**Comment by Evangelink** [comment]
@mavasani Is there anything you think is worth doing here? If not we could close the ticket.

**Comment by mavasani** [comment]
Yes, the rule is purely for FxCop migration and will be disabled/deprecated in .NET analyzers package. Closing it out.

**Comment by manfred-brands** [comment]
@mavasani Does this mean that public nested types in general are no longer discouraged or is there a replacement rule?
